controllers/goharbor/harborcluster/database/postgresql.go

```diff
@@ -1,15 +1,74 @@
 package database
 
 import (
+	"context"
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+
+	"github.com/go-logr/logr"
 	"github.com/goharbor/harbor-operator/apis/goharbor.io/v1alpha2"
+	goharborv1alpha2 "github.com/goharbor/harbor-operator/apis/goharbor.io/v1alpha2"
+	"github.com/goharbor/harbor-operator/pkg/k8s"
 	"github.com/goharbor/harbor-operator/pkg/lcm"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
 )
 
 type PostgreSQLController struct {
+	HarborCluster *goharborv1alpha2.HarborCluster
+	Ctx           context.Context
+	Client        k8s.Client
+	Log           logr.Logger
+	DClient       k8s.DClient
+	Scheme        *runtime.Scheme
+	ExpectCR      *unstructured.Unstructured
+	ActualCR      *unstructured.Unstructured
+}
+
+type Connect struct {
+	Host     string
+	Port     string
+	Password string
+	Username string
+	Database string
 }
 
 func (p *PostgreSQLController) Apply(harborcluster *v1alpha2.HarborCluster) (*lcm.CRStatus, error) {
-	panic("implement me")
+
+	p.Client.WithContext(p.Ctx)
+	p.DClient.WithContext(p.Ctx)
+	p.HarborCluster = harborcluster
+
+	crdClient := p.DClient.WithResource(databaseGVR).WithNamespace(p.HarborCluster.Namespace)
```

Comment:
How does it work when using external database

Reply:
If an external database is used, will this function be executed?

Reply:
Only HarborCluster CR defines InClusterDatabase will execute this func.